python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -196,235 +196,239 @@ def run(self, exposure, templateExposure, subtractedExposure, psfMatchingKernel,
                     np.all(np.isnan(templateExposure.getMaskedImage().getImage().getArray()))):
                 self.log.warn('Template or science image is entirely NaNs: skipping decorrelation.')
                 outExposure = subtractedExposure.clone()
-                return pipeBase.Struct(correctedExposure=outExposure, correctionKernel=None)
+                return pipeBase.Struct(correctedExposure=outExposure, )
+
+        tOverSVar = tvar/svar
+        if tOverSVar > 1e8:
```

Comment:
As the matching kernel high frequency components go to zero, the correction factor converges to sqrt(tvar/svar) i.e. this is a warning when high frequency pixels of the difference image is multiplied by more than 4 orders of magnitude as correction. Yes, in the regular case, the template variance is the smaller one and this is no concern; this can happen if we convolve the science image and thus svar becomes << tvar due to the convolution.

I don't really have a good justification for the value itself at the moment, just wanted an indicator to watch for diverging solutions, we'll see whether this is going to be met in practice.